### PR TITLE
Fix/cart not showing until refresh bug

### LINKED ISSUE #44: Cart doesn't show items unless manually refresh the page

### PR DESCRIPTION
**Pull request by mxng**
## Why

This pull request is needed because:

The cart does not show unless you refresh anywhere in the site

## What

This pull request introduces the following:

 * many appearance changes related to the cart
 * fix cart not showing unless refresh is done
 * cart quantity validation changed
 * instead of showing user invalid input, just don't allow user to input below 0
 * if user put 0, cart will show item but the price is not counted
 
## Other

I can't load the stripe page even with .env set up

### CODE REVIEW

**Review by tjw0612 — APPROVED** [review]
thanks for making the changes

File: app/views/food_orders/show.html.erb [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
 
 <div id="cart" class="d-none" style="background-color: white">
 
-  <div class="card-items">
-    <h1 class="mt-3"> Restaurant name </h1>
-    <p> Food items</p>
+  <div style="background-color: white">
+    <div class="container py-3">
+      <h1 class="mt-3">Last Step - Checkout</h1>
+      <p class="checkout-restaurant-name"></p>
```
Comment (TeeWhyKay) [inline app/views/food_orders/show.html.erb:14]: what is this empty p tag doing?

Comment (mxng) [reply app/views/food_orders/show.html.erb:14]: it's for me to do innerText that inserts the resto name in JS

File: app/javascript/plugins/init_cart.js [hunk 2]
```diff
@@ -35,13 +38,22 @@ const initCart = () => {
...
-    <p> ${item.dishName} - Price: $${item.dishPrice.cents / 100} </p>
+    <h5 class="card-title">${item.dishName}</h5>
+    <h6 class="card-subtitle mb-2 text-muted">Price: $${price} (Quantity: ${item.quantity})</h6>
+    <p>Subtotal: $${(price * item.quantity).toFixed(2)}</p>
     `
   });
-  cartItems.insertAdjacentHTML('beforeend', `<p>Total: $${data.total}</p>`)
+
+  const checkoutRestaurantName = cart.querySelector('.checkout-restaurant-name');
+  checkoutRestaurantName.innerHTML = `Restaurant: <i>${currentRestaurant}</i>`;
```
Comment (mxng) [inline app/javascript/plugins/init_cart.js:54]: @TeeWhyKay

File: app/views/food_orders/show.html.erb (hunk 1 above)
Comment (mxng) [reply app/views/food_orders/show.html.erb:14]: tagged you in the relevant line

**Review by TeeWhyKay — COMMENTED** [review]
in line 14 of app/views/food_orders/show.html.erb 
What is the empty P tag's purpose?

File: app/javascript/plugins/init_cart.js (hunk 2 above)
Comment (TeeWhyKay) [reply app/javascript/plugins/init_cart.js:54]: Oh nice.

**Pull request merged**